### PR TITLE
WT-6822 Use the correct python for all tests, and set up env for the split stress test.

### PR DESCRIPTION
**Pull request by ddanderson**

### CODE REVIEW

**Comment by ddanderson** [comment]
@keitharnoldsmith, could you do a quick review please?  This is fallout from removing Python.  I'm not sure why this worked before, but the split stress test needs environment set up so python can find the `_workgen.so` shared library.  At the same time, I noticed that /usr/bin/python3 was being run for that test, and we want to standardize on the toolchain python.

File: test/evergreen.yml [hunk 1]
```diff
@@ -2236,7 +2236,7 @@ tasks:
           script: |
             set -o errexit
             set -o verbose
-            for i in {1..10}; do ${python_binary|python3} split_stress.py; done
+            for i in {1..10}; do ${test_env_vars|} ${python_binary|python3} split_stress.py; done
```
Comment (sueloverso) [inline test/evergreen.yml:2239]: Just for my understanding, the `python_binary` part was already there. Below you add a lot of lines to set `python_binary`. Where was it coming from before?

**Review by sueloverso — APPROVED** [review]
LGTM. One question for my understanding.

File: test/evergreen.yml (hunk 1 above)
Comment (ddanderson) [reply test/evergreen.yml:2239]: Before, `python_binary` was set only for Windows.  So, before this change, it was defaulting to `python3` on Unix, which would get the toolchain python only if the PATH was set correctly.

**Pull request merged**